src/components/ControlDropdown/ControlDropdown.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { isNotUndefined } from 'ramda-adjunct';
+
+import { useDropdown } from '../../hooks/useDropdown';
+import { SpaceSizes } from '../../theme';
+import { SSCIconNames } from '../../theme/icons/icons.enums';
+import { pxToRem } from '../../utils';
+import { Inline, Padbox, Stack } from '../layout';
+import { Button } from '../Button';
+import {
+  ButtonColors,
+  ButtonSizes,
+  ButtonVariants,
+} from '../Button/Button.enums';
+import { Icon } from '../Icon';
+import { H4 } from '../typography';
+import { ControlDropdownProps } from './ControlDropdown.types';
+import { ControlDropdownPlacements } from './ControlDropdown.enums';
+
+const CloseButton = styled(Button)`
+  padding: 0;
+  width: ${pxToRem(16)};
+  height: ${pxToRem(16)};
+`;
+
+const ControlDropdown: React.FC<ControlDropdownProps> = ({
+  children,
+  title,
+  parentRef,
+  isOpen = false,
+  onOpen,
+  onClose,
+  onReset,
+  onSubmit,
+  submitLabel = 'Apply',
+  closeLabel = 'Close',
+  resetLabel = 'Reset to defaults',
+  placement = ControlDropdownPlacements.bottomLeft,
+  width = 370,
+}) => {
+  const { isPaneDisplayed, Pane } = useDropdown(parentRef, {
+    defaultIsPaneDisplayed: isOpen,
+    paneWidth: width,
+    onClickOut: onClose,
+    onShow: () => {
+      if (isPaneDisplayed) {
+        onClose();
+      } else {
+        onOpen();
+      }
+    },
+    onHide: onClose,
+    placement,
+    isElevated: true,
+    hasInternalShowHide: false,
+  });
+
+  return (
+    isOpen && (
```

Comment:
just to save indentation and readability might be worth checking this outside the JSX context, since it's a top level conditional